onnxruntime/core/providers/coreml/builders/impl/conv_op_builder.cc

```diff
@@ -46,25 +46,72 @@ Status ConvOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
   std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> layer = CreateNNLayer(node);
 
   const auto& input_defs = node.InputDefs();
+  const auto& output_defs = node.OutputDefs();
+  const auto& input_name = input_defs[0]->Name();
+  const auto& output_name = output_defs[0]->Name();
 
   const auto& weight_tensor = *model_builder.GetInitializerTensors().at(input_defs[1]->Name());
   const auto& weight_shape = weight_tensor.dims();
 
+  const bool is_1d_conv = (weight_shape.size() == 3);
+
   NodeAttrHelper helper(node);
   const auto strides = helper.Get("strides", std::vector<int64_t>{1, 1});
-  const auto onnx_pads = helper.Get("pads", std::vector<int64_t>{0, 0, 0, 0});
   const auto dilations = helper.Get("dilations", std::vector<int64_t>{1, 1});
+  const auto onnx_pads = helper.Get("pads", std::vector<int64_t>{0, 0, 0, 0});
+  // Strides/dilations for 1d conv is normally of length 1. Expand them by 1
+  // to meet the required length 2 (for 2d conv it's normally 2)
+  // Similarly 1d conv normally has a length 2 padding. Expand it to length 4 by adding additional zeros.
+  auto strides_prime = strides;
```

Comment:
Why do you need to create the `_prime` version of all these attributes, you can augment the original attributes, by adding missing fields right?
Since the end result is it will work as 2d conv
Say for 1d conv we get strides as {1}, we add extra 1 to it and make it {1,1} instead of create a `strides_prime`
Such that you don't have to use `  if (is_1d_conv) {...} else {...}` below

Reply:
The reason is still the attributes are all marked as `const`, so I have to make copy to modify them.

Reply:
We can remove `const` for those attributes

Reply:
oh.. but I saw the `.Get()` function is actually marked `const`, and it's from `onnxruntime/core/providers/shared/utils/utils.cc`. Is it also allowed to modify?